src/tyr/route_serializer_osrm.cc

```diff
@@ -418,19 +418,36 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
 
     // TODO: add recosted durations to the intersection?
 
-    // Get bearings and access to outgoing intersecting edges. Do not add
-    // any intersecting edges for the first depart intersection and for
-    // the arrive step.
-    std::vector<IntersectionEdges> edges;
+    // Add rest_stops when passing by a rest_area or service_area
     if (i > 0 && !arrive_maneuver) {
+      std::vector<std::string> rest_stop_types;
       for (uint32_t m = 0; m < node->intersecting_edge_size(); m++) {
         auto intersecting_edge = node->GetIntersectingEdge(m);
         bool routeable = intersecting_edge->IsTraversableOutbound(curr_edge->travel_mode());
-        uint32_t bearing = static_cast<uint32_t>(intersecting_edge->begin_heading());
-        edges.emplace_back(bearing, routeable, false, false);
+
+        if (routeable && intersecting_edge->use() == TripLeg_Use_kRestAreaUse) {
+          rest_stop_types.push_back("rest_area");
```

Comment:
Don't merge yet -- have some updates to make to this serialization

Reply:
ok

Reply:
cc @kdiluca